File: backend/routers/admin_dashboard.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session
from dependencies import get_current_admin_user, get_db
from models import User
from pydantic import BaseModel

router = APIRouter(prefix="/admin-dashboard", tags=["admin-dashboard"])

# 仪表盘数据模型
class DashboardData(BaseModel):
    total_users: int
    active_users: int
    api_bound_users: int
    total_trading_volume: float
    total_profit: float
    total_withdrawals: float
    total_commissions: float

# 获取仪表盘数据（仅管理员）
@router.get("/dashboard", response_model=DashboardData)
def get_dashboard_data(
    user_id: int,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_admin_user),
):
    total_users = db.query(User).count()
    active_users = db.query(User).filter(User.is_active == 1).count()
    api_bound_users = 50  # 示例数据
    total_trading_volume = 1000000.0  # 示例数据
    total_profit = 50000.0  # 示例数据
    total_withdrawals = 20000.0  # 示例数据
    total_commissions = 10000.0  # 示例数据

    return DashboardData(
        total_users=total_users,
        active_users=active_users,
        api_bound_users=api_bound_users,
        total_trading_volume=total_trading_volume,
        total_profit=total_profit,
        total_withdrawals=total_withdrawals,
        total_commissions=total_commissions,
    )

# 获取所有用户信息
@router.get("/users")
def get_all_users(db: Session = Depends(get_db)):
    users = db.query(User).all()
    return users

# 获取交易数据（示例）
@router.get("/transactions")
def get_transactions():
    # 示例：返回假数据
    return [
        {"id": 1, "user_id": 1, "symbol": "BTC/USDT", "amount": 0.1, "profit": 50},
        {"id": 2, "user_id": 2, "symbol": "ETH/USDT", "amount": 1, "profit": 100},
    ]

# 获取推广数据（示例）
@router.get("/referrals")
def get_referrals():
    # 示例：返回假数据
    return [
        {"user_id": 1, "referrals": 5, "commission": 200},
        {"user_id": 2, "referrals": 3, "commission": 120},
    ]

File: frontend/pages/admin-dashboard.js
import { useEffect, useState } from 'react';
import axios from 'axios';
import styles from '../styles/AdminDashboard.module.css';

export default function AdminDashboard() {
  const [dashboardData, setDashboardData] = useState({});
  const [users, setUsers] = useState([]);
  const [transactions, setTransactions] = useState([]);
  const [referrals, setReferrals] = useState([]);

  useEffect(() => {
    // 获取仪表盘数据
    axios.get('/admin-dashboard/dashboard').then((response) => setDashboardData(response.data));
    // 获取用户数据
    axios.get('/admin-dashboard/users').then((response) => setUsers(response.data));
    // 获取交易数据
    axios.get('/admin-dashboard/transactions').then((response) => setTransactions(response.data));
    // 获取推广数据
    axios.get('/admin-dashboard/referrals').then((response) => setReferrals(response.data));
  }, []);

  return (
    <div className={styles.container}>
      <h1>管理员后台</h1>

      <section className={styles.dashboard}>
        <h2>仪表盘数据</h2>
        <p>总用户数: {dashboardData.total_users}</p>
        <p>活跃用户数: {dashboardData.active_users}</p>
        <p>API 绑定用户数: {dashboardData.api_bound_users}</p>
        <p>累计交易金额: ${dashboardData.total_trading_volume}</p>
        <p>累计收益: ${dashboardData.total_profit}</p>
        <p>累计提现: ${dashboardData.total_withdrawals}</p>
        <p>累计佣金: ${dashboardData.total_commissions}</p>
      </section>

      <section>
        <h2>用户数据</h2>
        <table className={styles.table}>
          <thead>
            <tr>
              <th>ID</th>
              <th>邮箱</th>
              <th>状态</th>
              <th>注册时间</th>
            </tr>
          </thead>
          <tbody>
            {users.map((user) => (
              <tr key={user.id}>
                <td>{user.id}</td>
                <td>{user.email}</td>
                <td>{user.is_active ? '活跃' : '禁用'}</td>
                <td>{user.created_at}</td>
              </tr>
            ))}
          </tbody>
        </table>
      </section>

      <section>
        <h2>交易数据</h2>
        <table className={styles.table}>
          <thead>
            <tr>
              <th>ID</th>
              <th>用户ID</th>
              <th>交易对</th>
              <th>数量</th>
              <th>收益</th>
            </tr>
          </thead>
          <tbody>
            {transactions.map((tx) => (
              <tr key={tx.id}>
                <td>{tx.id}</td>
                <td>{tx.user_id}</td>
                <td>{tx.symbol}</td>
                <td>{tx.amount}</td>
                <td>{tx.profit}</td>
              </tr>
            ))}
          </tbody>
        </table>
      </section>

      <section>
        <h2>推广数据</h2>
        <table className={styles.table}>
          <thead>
            <tr>
              <th>用户ID</th>
              <th>邀请人数</th>
              <th>佣金</th>
            </tr>
          </thead>
          <tbody>
            {referrals.map((ref) => (
              <tr key={ref.user_id}>
                <td>{ref.user_id}</td>
                <td>{ref.referrals}</td>
                <td>{ref.commission}</td>
              </tr>
            ))}
          </tbody>
        </table>
      </section>
    </div>
  );
}
